### PR TITLE
PorousFlow: JOSS bibliography update

### LINKED ISSUE #16312: PorousFlow: JOSS paper bib changes
## Reason

JOSS editors require some minor changes: see https://github.com/openjournals/joss-reviews/issues/2176 .

## Design

Change a date and add a URL in `porous_flow/paper.bib`

### PR DESCRIPTION
**Pull request by WilkAndy**
Fixes #16312

### CODE REVIEW

File: modules/porous_flow/paper.bib [hunk 1]
```diff
@@ -22,6 +22,7 @@ @TechReport{pflotran
 Author = "Peter C. Lichtner and Glenn E. Hammond and Chuan Lu and Satish Karra and Gautam Bisht and Benjamin Andre and Richard T. Mills and Jitendra Kumar and Jennifer M. Frederick",
 Title = "{PFLOTRAN} {U}ser {M}anual",
 Note = "http://documentation.pflotran.org",
+url = "http://documentation.pflotran.org",
```
Comment (jedbrown) [inline modules/porous_flow/paper.bib:25]:
```suggestion
url = "https://documentation.pflotran.org",
```
Might as well so modern browsers don't complain. (Soon, they should all attempt to promote.)

Comment (WilkAndy) [reply modules/porous_flow/paper.bib]: Fixed, thanks

**Comment by WilkAndy** [comment]
The external app "Okapi" has problems unrelated to this PR

**Comment by WilkAndy** [comment]
Hi @loganharbour or @aeslaughter .   I think you'll agree this minor change can be merged.

**Review by loganharbour — APPROVED** [review]
Next is failing right now - I'd prefer to wait until we get that fixed to get this in.

**Comment by WilkAndy** [comment]
Yea, sure, @loganharbour , there's no huge urgency here

**Pull request merged**